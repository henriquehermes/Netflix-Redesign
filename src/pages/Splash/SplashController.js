import React, { useEffect } from 'react';
import AsyncStorage from '@react-native-community/async-storage';
import SplashComponent from './SplashComponent';

export default function SplashController({ navigation }) {
  useEffect(() => {
    async function getUserLogged() {
      const response = await AsyncStorage.getItem('User');
      if (response) {
        navigation.navigate('AppNavigator');
      } else {
        navigation.navigate('AuthNavigator');
      }
    }
    getUserLogged();
  }, []);

  return <SplashComponent />;
}
