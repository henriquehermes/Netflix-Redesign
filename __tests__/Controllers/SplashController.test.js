/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import SplashController from '../../src/pages/Splash/SplashController';

const fakeLoginInApp = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(<SplashController navigation={fakeLoginInApp} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
