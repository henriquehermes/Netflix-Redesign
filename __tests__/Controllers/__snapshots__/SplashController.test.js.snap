// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#111111",
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/img/Netflix_Logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 100,
        },
      ]
    }
  />
  <ActivityIndicator
    animating={true}
    color="#D7240F"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
